app/models/article.rb

```diff
@@ -409,7 +409,7 @@ def set_caches
 
     self.cached_user_name = user_name
     self.cached_user_username = user_username
-    self.path = calculated_path
+    self.path = calculated_path.downcase
```

Comment:
Moved the `.downcase` call here (to the `before_save` callback). This way we won't run into upcase usernames slipping through into Article paths.